### PR TITLE
[ADI] Add swagger support

### PR DESCRIPTION
**Pull request by Remg**

### CODE REVIEW

File: app/config/services/services.xml [hunk 1]
```diff
@@ -402,5 +402,15 @@
...
         </service>
+
+        <!-- Swagger -->
+        <service id="AppBundle\Swagger\SwaggerDecorator"
+            decorates="api_platform.swagger.normalizer.documentation"
+            autoconfigure="false">
+            <argument key="$decorated" type="service" id="AppBundle\Swagger\SwaggerDecorator.inner"/>
+            <argument key="$apiPathPrefix">%api_path_prefix%</argument>
+            <argument key="$operationPathResolver" type="service" id="api_platform.operation_path_resolver"/>
+            <argument key="$paginationEnabled">%%</argument>
```
Comment (ottaviano) [inline app/config/services/services.xml]: what is it? :)

File: features/api/ideas_workshop_ideas.feature [hunk 2]
```diff
@@ -1,4 +1,5 @@
 @api
+  @ideas
```
Comment (ottaviano) [inline features/api/ideas_workshop_ideas.feature]: remove it?

File: app/config/services/services.xml (hunk 1 above)
Comment (Remg) [reply app/config/services/services.xml]: glory

File: src/Entity/IdeasWorkshop/Idea.php [hunk 3]
```diff
@@ -29,7 +29,51 @@
...
+ *                         "type": "string",
+ *                         "description": "The theme name of the Idea resource.",
+ *                         "example": "armée",
+ *                     },
+ *                     {
+ *                         "name": "author_category",
+ *                         "in": "query",
+ *                         "type": "uuid",
+ *                         "description": "The author category of the Idea resource.",
+ *                         "enum": AuthorCategoryEnum::ALL_CATEGORIES,
```
Comment (illusionOfParadise) [inline src/Entity/IdeasWorkshop/Idea.php]: No need to create another constant, you can do `AuthorCategoryEnum::toArray()` to get all categories.

File: src/Entity/IdeasWorkshop/Idea.php [hunk 4]
```diff
@@ -29,7 +29,51 @@
...
+ *             "method": "GET",
+ *             "requirements": {"id": "%pattern_uuid%"},
+ *             "swagger_context": {
+ *                 "parameters": {
+ *                     {
+ *                         "name": "status",
+ *                         "in": "query",
+ *                         "type": "string",
+ *                         "description": "The status of the Idea resource.",
+ *                         "enum": IdeaStatusEnum::ALL_STATUSES,
```
Comment (illusionOfParadise) [inline src/Entity/IdeasWorkshop/Idea.php]: Same here, you can use IdeaStatusEnum::toArray().

File: src/Entity/IdeasWorkshop/Idea.php (hunk 3 above)
Comment (Remg) [reply src/Entity/IdeasWorkshop/Idea.php]: i tried this before, but callbacks are not handled here. only a constant would do the trick

File: src/Entity/IdeasWorkshop/IdeaStatusEnum.php [hunk 5]
```diff
@@ -15,4 +15,11 @@
         self::PENDING,
         self::FINALIZED,
     ];
+
+    public const ALL_STATUSES = [
```
[the remark this answers is not in the document]
Comment (Remg) [reply src/Entity/IdeasWorkshop/IdeaStatusEnum.php]: i need a constant to use in my annotations.
callbacks are not handled in swagger documentation.

File: src/Entity/IdeasWorkshop/Idea.php (hunk 4 above)
Comment (Remg) [reply src/Entity/IdeasWorkshop/Idea.php]: as above: callbacks are not supported here.

File: src/Entity/IdeasWorkshop/AuthorCategoryEnum.php [hunk 6]
```diff
@@ -9,4 +9,10 @@
     public const COMMITTEE = 'COMMITTEE';
     public const ADHERENT = 'ADHERENT';
     public const QG = 'QG';
+
+    public const ALL_CATEGORIES = [
```
[the remark this answers is not in the document]
Comment (Remg) [reply src/Entity/IdeasWorkshop/AuthorCategoryEnum.php]: nice catch.
as i need a constant, i will stop extending Enum

**Pull request merged**